src/core/tools.js

```diff
@@ -412,7 +414,38 @@ exports.getPrimaryKey = (options) => R.pathOr('id', ['churros', 'primaryKey'], o
 
 /**
  * Validate payload against JSON schema
- * @param {Object} body 
- * @param {Object} schema 
+ * @param {Object} body
+ * @param {Object} schema
  */
-exports.isValidSchema = (body, schema) => tv4.validate(body, schema, (isValid, validationError) => !validationError && isValid);
\ No newline at end of file
+exports.isValidSchema = (body, schema) => tv4.validate(body, schema, (isValid, validationError) => !validationError && isValid);
+/*
+   This method is to take care of finding value even if primary key is nested
+*/
+exports.getPrimaryKeyValue = (primaryKey, body) => {
+  let nestedKeys = primaryKey.split('.');
+  let createdId = R.path(nestedKeys, body);
+  return createdId;
+};
+
+exports.constructChurrosOptionsFromAllArgs = (createPayload, updatePayload, primaryKey) => {
+  const options = {
+    churros: {
+      updatePayload: updatePayload,
+      primaryKey: primaryKey
+    },
+    payload: createPayload
+  };
+  console.log('Actual value build is ' + JSON.stringify(options));
+  return options;
+};
+
+exports.constructChurrosOptionsFromMainPayload = (testDetails) => {
+  const options = {
```

Comment:
It might be worth just using your above function, i.e. 
```suggestion
 return constructChurrosOptionsFromAllArgs(testDetails.createPayload, testDetails.updatePayload, testDetails.primaryKey);
```